### DISCUSSION TITLE
How to remove gap between graph element

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#918] [Q&A] [answered]
Hi,
   I create a node containing three elements 'left', 'center', and 'right'. There is a problem that confuses me. There is always some gap between the three parts, I do not know how to remove this gap.
![image](https://user-images.githubusercontent.com/4537809/218498434-d0525593-903a-4fd1-b952-8508c7d16c96.png)
I hope it looks like follow picture, with no gap between them
![image](https://user-images.githubusercontent.com/4537809/218498364-b17df965-2754-4ed7-8da5-a9243a457be8.png)
I do not add any margin or padding property in css. I do not set hGap in the layout options

**Comment by CamilleLetavernier** [comment, marked as answer]
Hi,

The default layout options for hbox and vbox include a non-0 gap, as well as some padding. If you're using one of these layouts, you may need to explicitly set these values to 0 in the layout options. Child nodes may also have their own padding, which may or may not need to be set to 0.

```ts
protected override getDefaultLayoutOptions(): HBoxLayoutOptionsExt {
        return {
            resizeContainer: true,
            paddingTop: 5,
            paddingBottom: 5,
            paddingLeft: 5,
            paddingRight: 5,
            paddingFactor: 1,
            hGap: 1,
[...]
```

**Reply by wss29** [reply]
Thank you very much.
